### PR TITLE
Add rspec example flag

### PR DESCRIPTION
**Pull request by AspenJames**
Add support for using RSpec's `--example` flag
@aviflombaum 
#Staff

### CODE REVIEW

File: bin/learn-test [hunk 1]
```diff
@@ -57,6 +57,10 @@ OptionParser.new do |opts|
     options[:fail_fast] = f
   end
 
+  opts.on("-e", "--example STRING", "Run examples whose full nested names include STRING (may be used more than once)") do |s|
```
Comment (aviflombaum) [inline bin/learn-test]: So while rspec natively supports [multiple uses of `--example`](https://relishapp.com/rspec/rspec-core/v/3-8/docs/command-line/example-option#multiple-applications-of-example-name-option) flags, the implementation here does not I don't think.

Comment (aviflombaum) [reply bin/learn-test]: I've added this test

```ruby
    it 'accepts multiple examples' do
      output = `cd ./spec/fixtures/rspec-unit-spec && ./../../../bin/learn-test --local --test --example multiple --example accepts`

      expect(output).to include('2 examples, 0 failures')
      expect(output).to_not include('3 examples')
      expect(output).to_not include('1 example')
    end    
```

and updated the fixture:

```ruby
  it 'has multiple specs' do
    expect(subject).not_to be_an_instance_of(String)
  end

  it 'has accepts more than one example' do
    expect(subject).not_to be_an_instance_of(String)
  end  
```

to confirm and it fails. looking into how to glob multiple `--examples` through `OptionParser`

File: bin/learn-test [hunk 2]
```diff
@@ -57,6 +57,10 @@ OptionParser.new do |opts|
...
+    options[:example] = s
```
Comment (aviflombaum) [inline bin/learn-test]:
```ruby
options[:example] ||= []
options[:example] << s
```

Will do it along with the next suggestion.

File: lib/learn_test/strategies/rspec.rb [hunk 3]
```diff
@@ -25,6 +25,10 @@ def configure
           argv << '--fail-fast'
         end
 
+        if example_option_present?
+          argv << "--example #{options[:example]}"
```
Comment (aviflombaum) [inline lib/learn_test/strategies/rspec.rb]:
```ruby
argv << "#{options[:example].collect{|e| "--example #{e}"}.join(" ")}"
```

Comment (aviflombaum) [reply lib/learn_test/strategies/rspec.rb]: The only thing I don't like about this is it assumes `options[:example]` is an array but that's okay as we set it earlier upon detection of the `--example` flag. We could be more defensive but I'd be okay with that.

File: bin/learn-test (hunk 2 above)
Comment (aviflombaum) [reply bin/learn-test]: If you wanted to be really fancy:

```ruby
(options[:example] ||= []) << s
```

**Comment by AspenJames** [comment]
Requested changes made - thanks for catching that @aviflombaum!

File: lib/learn_test/strategies/rspec.rb (hunk 3 above)
Comment (AspenJames) [reply lib/learn_test/strategies/rspec.rb]: We could check the class type of `options[:example]` here and perform different actions?

Even `[options[:example]].flatten` would work but that's ugly

**Comment by aviflombaum** [comment]
@meryldakin @gj as you have open work on this gem, are you good merging this as you resolve your PR for jest and cut a new learn-test gem and learn gem? Not sure what our version release plans are but I reviewed @AspenJames code here and it's good, seems unlikely to cause any issues and is tested.

**Comment by AspenJames** [comment]
Done!

**Comment by meryldakin** [comment]
yep sounds good @aviflombaum!

**Comment by AspenJames** [comment]
@gj can I get an update on the status of this?

**Comment by gj** [comment]
Hi @AspenJames,

I don't know what the status of #57 is; perhaps @meryldakin knows. I'm happy to ship this PR the next time I need to update the gem, but I'm pretty slammed with other things at the moment.

**Comment by aviflombaum** [comment]
I re-reviewed and test this and am comfortable merging this - haven't checked #57 but that's def more complex, but this I would be down to merge. If @meryldakin and @gj agree, let's just merge and well plan a new cut of the gem soon but at least get this code in there?

**Comment by gj** [comment]
@aviflombaum sure, it's a harmless change even if the next person to cut a release of the gem doesn't realize this new code has been merged in.

**Comment by AspenJames** [comment]
Thanks all!

**Pull request merged**